### ISSUE TITLE
Add expression attribute for repl display

### ISSUE THREAD

**Issue by leonardschneider** [diku-dk/futhark#1567]
I like the Futhark repl, but I'm missing formatting while printing the values. I propose adding some basic formatting, which can come handy while debugging a program, by adding expression attributes that will be interpreted by the repl to display expressions.

Here are some examples I'd like to have.

```
> #[char] 97u8 -- display a character
'a'
> #[utf8] "Hello, world!" -- display a string in utf8 encoding
"Hello, world"
> #[raw] "Hello, world!" -- direct display for extensibility
Hello, world! 
```

**Comment by athas** [comment]
I don't think an expression is the right way to do this.  What about a `:string` command that prints the result of the expression as a string literal (if it has an appropriate type)?

**Comment by FluxusMagna** [comment]
That could probably also be useful in futhark literate.

**Comment by athas** [comment]
Yeah, now that I think about it, maybe a general `:format` command that accepts a `printf`-style format string would be better.  The same machinery could be used for `futhark literate`.

**Comment by leonardschneider** [comment]
You're right. I've played with a prototype and a command makes more sense.

**Comment by athas** [comment]
Instead of `printf` style, I think the most elegant option would be something similar to [Pythons formatted string literals](https://docs.python.org/3/tutorial/inputoutput.html#formatted-string-literals).  They are similar to `printf` strings, but instead of being positional, you put the name (or eventually, expression) of what you want to print:

```
> let str = "foo"
> :print "{str}"
[102u8, 111u8, 111u8]
> :print "{str:s}"
"foo"
```

With no specifier it will print in the default manner for the type, but with a specifier we can force e.g. arrays to be interpreted as UTF-8 strings.
